model/atmosphere/dycore/tests/dycore_tests/test_velocity_advection.py

```diff
@@ -394,31 +410,691 @@ def test_velocity_corrector_step(
         h_grid.domain(dims.EdgeDim)(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_7)
     )
     assert helpers.dallclose(
-        velocity_advection.z_v_grad_w.asnumpy()[start_cell_lateral_boundary_level_7:, :],
+        velocity_advection._khalf_horizontal_advection_of_w_at_edge.asnumpy()[
+            start_cell_lateral_boundary_level_7:, :
+        ],
         icon_result_z_v_grad_w[start_cell_lateral_boundary_level_7:, :],
         atol=1e-16,
     )
     # stencil 08
     start_cell_nudging = icon_grid.start_index(h_grid.domain(dims.CellDim)(h_grid.Zone.NUDGING))
     assert helpers.dallclose(
-        velocity_advection.z_ekinh.asnumpy()[start_cell_nudging:, :],
+        velocity_advection._horizontal_kinetic_energy_at_cell.asnumpy()[start_cell_nudging:, :],
         savepoint_velocity_exit.z_ekinh().asnumpy()[start_cell_nudging:, :],
     )
 
     # stencil 11,12,13,14
     assert helpers.dallclose(
-        velocity_advection.z_w_con_c.asnumpy()[start_cell_nudging:, :],
+        velocity_advection._khalf_contravariant_corrected_w_at_cell.asnumpy()[
+            start_cell_nudging:, :
+        ],
         savepoint_velocity_exit.z_w_con_c().asnumpy()[start_cell_nudging:, :],
     )
     # stencil 16
     assert helpers.dallclose(
-        diagnostic_state.ddt_w_adv_pc.corrector.asnumpy()[start_cell_nudging:, :],
+        diagnostic_state.vertical_wind_advective_tendency.corrector.asnumpy()[
+            start_cell_nudging:, :
+        ],
         icon_result_ddt_w_adv_pc[start_cell_nudging:, :],
         atol=5.0e-16,
     )
     # stencil 19
     assert helpers.dallclose(
-        diagnostic_state.ddt_vn_apc_pc.corrector.asnumpy(),
+        diagnostic_state.normal_wind_advective_tendency.corrector.asnumpy(),
         icon_result_ddt_vn_apc_pc,
         atol=5.0e-16,
     )
+
+
+@pytest.mark.dataset
+@pytest.mark.embedded_remap_error
+@pytest.mark.parametrize(
+    "experiment, step_date_init, step_date_exit",
+    [
+        (dt_utils.REGIONAL_EXPERIMENT, "2021-06-20T12:00:10.000", "2021-06-20T12:00:10.000"),
+        (dt_utils.GLOBAL_EXPERIMENT, "2000-01-01T00:00:02.000", "2000-01-01T00:00:02.000"),
+    ],
+)
+@pytest.mark.parametrize("istep_init", [1])
+@pytest.mark.parametrize("substep_init", [1])
+def test_velocity_fused_1_7_compute_edge_diagnostics_for_velocity_advection_in_predictor_step(
```

Comment:
If you test a function `def func` I would call the test simply `test_func`
Whether it used to be call `fused_n_m` is completely irrelevant, and whether it is predictor or correct is given by the value of the `istep_init` or `istep_exit` fixture.  If the function only is for predictor then these have value 1 like here, if they are called in both of them you can truly parametrize.